cassovary-collections/src/main/scala/com/twitter/cassovary/collections/SortedCSeqOps.scala

```diff
@@ -80,13 +96,8 @@ object SortedArrayOps {
       }
     }
 
-    slice(union, 0, resultIdx)
-  }
-
-  @inline def slice(a: Array[Int], from: Int, len: Int): Array[Int] = {
-    val result = Array.ofDim[Int](len)
-    Array.copy(a, from, result, 0, len)
-    result
+    // possible memory leak
```

Comment:
The CSeq we create in the next line is a slice of array defined in line 34 that has length: min(array.length, that.length). Cells from intersectionIdx to the end are wasted. 

Alternatively we can use ArrayBuffer from fastutils (that is not ready yet) or copy indices before intersectionIdx to a new array (which should be equivalent). In this case we waste speed to gain some memory. What do you think about this tradeoff?